File: beecrowd/cpp/p1021.cpp
// Notas e Moedas

#include <bits/stdc++.h>

using namespace std;

int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    double t;
    int cents;
    int nts[] = {10000, 5000, 2000, 1000, 500, 200};
    int mds[] = {100, 50, 25, 10, 5, 1};
    cin >> t;
    cents = round(t * 100);
    cout << "NOTAS:\n";
    for (int nt : nts)
    {
        cout << cents / nt << " nota(s) de R$ " << nt / 100 << ".00\n";
        cents = cents % nt;
    }

    cout << "MOEDAS:\n";
    for (int md : mds)
    {
        cout << cents / md << fixed << setprecision(2) << " moeda(s) de R$ " << md / 100.00 << '\n';
        cents = cents % md;
    }
    return 0;
}

File: beecrowd/python/p1021.py
# Notas e Moedas

import sys

cents = int(float(sys.stdin.read()) * 100 + 0.5)

n100, cents = divmod(cents, 10000)
n50, cents = divmod(cents, 5000)
n20, cents = divmod(cents, 2000)
n10, cents = divmod(cents, 1000)
n5, cents = divmod(cents, 500)
n2, cents = divmod(cents, 200)

m1, cents = divmod(cents, 100)
m50, cents = divmod(cents, 50)
m25, cents = divmod(cents, 25)
m10, cents = divmod(cents, 10)
m5, cents = divmod(cents, 5)

sys.stdout.write(
    f"NOTAS:\n"
    f"{n100} nota(s) de R$ 100.00\n"
    f"{n50} nota(s) de R$ 50.00\n"
    f"{n20} nota(s) de R$ 20.00\n"
    f"{n10} nota(s) de R$ 10.00\n"
    f"{n5} nota(s) de R$ 5.00\n"
    f"{n2} nota(s) de R$ 2.00\n"
    f"MOEDAS:\n"
    f"{m1} moeda(s) de R$ 1.00\n"
    f"{m50} moeda(s) de R$ 0.50\n"
    f"{m25} moeda(s) de R$ 0.25\n"
    f"{m10} moeda(s) de R$ 0.10\n"
    f"{m5} moeda(s) de R$ 0.05\n"
    f"{cents} moeda(s) de R$ 0.01\n"
)
